### PR TITLE
Clean up built-in types

### PR DESCRIPTION
**Pull request by ttuegel**
Continuation of #2209.

- Pull the built-in primitive types out of `Builtin`:
  - `Kore.Internal.InternalInt`
  - `Kore.Internal.InternalBool`
  - `Kore.Internal.InternalString`
- Split up the simplification routines for each type.
- Reduce some duplication in the unification code.
- Add missing test cases for unification.
- Update test output. The changes should only be to comments.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [x] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [x] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [x] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [x] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/Kore/Internal/TermLike.hs [hunk 1]
```diff
@@ -1721,11 +1757,11 @@ pattern Builtin_
     -> TermLike variable
 
 pattern BuiltinBool_
-    :: Domain.InternalBool
+    :: InternalBool
     -> TermLike variable
 
-pattern BuiltinInt_
-    :: Domain.InternalInt
+pattern InternalInt_
```
Comment (ana-pantilie) [inline kore/src/Kore/Internal/TermLike.hs:1763]: Why was only `BuiltinInt` renamed?

Comment (ttuegel) [reply kore/src/Kore/Internal/TermLike.hs:1763]: Thanks for pointing this out! I simply forgot the others.

File: kore/src/Kore/Internal/InternalBool.hs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+-- | An 'InternalBool' pattern is always 'Function'.
+instance Synthetic Function (Const InternalBool) where
+    synthetic = const (Function True)
+    {-# INLINE synthetic #-}
+
+-- | An 'InternalBool' pattern is always 'Functional'.
+instance Synthetic Functional (Const InternalBool) where
+    synthetic = const (Functional True)
+    {-# INLINE synthetic #-}
+
```
[the remark this answers is not in the document]
Comment (ttuegel) [reply kore/src/Kore/Internal/InternalBool.hs:88]: I think you're right, I will leave off the comments unless some instance is unusual.

**Comment by ttuegel** [comment]
@ana-pantilie I addressed you comments, could you please take another look?

**Review by ana-pantilie — APPROVED** [review]
Looks good to me. I added a commit with some test cases (it used to be a TODO).

**Comment by ttuegel** [comment]
@ana-pantilie Oh, I added the same sort of tests at `Test.Kore.Builtin.{Bool,String}.test_contradiction`. I will consolidate them.

**Comment by ana-pantilie** [comment]
@ttuegel Sorry, I saw the TODO and thought I could add them quickly, it probably just means the TODO was out of date. You can just revert the commit if you think they're redundant and it's easier that way.

**Pull request merged**